### DISCUSSION TITLE
AUCall cannot be output

### DISCUSSION THREAD

**Discussion by jeslo328** [humanpred/pknca#309] [Show and tell]
I have a data that has 2 periods. And the sampling is till 10 hours. I can see the tlast for eg subject 1 below.
[I tried end = 10 too…[
Upon using the PKNCA I am not able to output the auclast. Below are my code and output sample.
Appreciate if you could help me navigate.
Can I schedule few minute meet. If it would help ?

```
#---------------------#
o_conc <- PKNCAconc(my_data1,PKCONC~HOUR|as.factor(Treatment)+as.factor(SUBJID))
o_dose <- PKNCAdose(my_rand_dose_all,Dose~HOUR|as.factor(Treatment)+ as.factor(SUBJID))


data_obj <- PKNCAdata(o_conc, o_dose,
                      intervals=data.frame(start=0, 
                                           end=Inf,
                                           cmax=T,
                                           tmax=T,
                                           tlast=T,
                                           half.life=T,
                                           aucall=TRUE,
                                           auclast=TRUE,
                                           aucinf.pred=T,
                                           aucinf.obs=T,
                                           cl.obs=T,
                                           vd.obs=T))

SUBJID	PERIOD	DAY	HOUR	MATRIX	ANALYTE	CONC	LLOQ	UNITS	ARMCD	Treatment
1	1	1	-1	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	1	1	0.25	Plasma	Alendronate	160.74	.	pg/mL	BA	Alendronate + 016
1	1	1	0.5	Plasma	Alendronate	447.7	.	pg/mL	BA	Alendronate + 016
1	1	1	0.75	Plasma	Alendronate	584.31	.	pg/mL	BA	Alendronate + 016
1	1	1	1	Plasma	Alendronate	594.39	.	pg/mL	BA	Alendronate + 016
1	1	1	1.5	Plasma	Alendronate	705.83	.	pg/mL	BA	Alendronate + 016
1	1	1	2	Plasma	Alendronate	439.72	.	pg/mL	BA	Alendronate + 016
1	1	1	3	Plasma	Alendronate	181.16	.	pg/mL	BA	Alendronate + 016
1	1	1	4	Plasma	Alendronate	138.6	.	pg/mL	BA	Alendronate + 016
1	1	1	5	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	1	1	6	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	1	1	7	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	1	1	8	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	1	1	10	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate + 016
1	2	1	-1	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate 
1	2	1	0.25	Plasma	Alendronate	127.24	.	pg/mL	BA	Alendronate 
1	2	1	0.5	Plasma	Alendronate	585.54	.	pg/mL	BA	Alendronate 
1	2	1	0.75	Plasma	Alendronate	857.92	.	pg/mL	BA	Alendronate 
1	2	1	1	Plasma	Alendronate	907.85	.	pg/mL	BA	Alendronate 
1	2	1	1.5	Plasma	Alendronate	864.7	.	pg/mL	BA	Alendronate 
1	2	1	2	Plasma	Alendronate	791.29	.	pg/mL	BA	Alendronate 
1	2	1	3	Plasma	Alendronate	759.78	.	pg/mL	BA	Alendronate 
1	2	1	4	Plasma	Alendronate	562.44	.	pg/mL	BA	Alendronate 
1	2	1	5	Plasma	Alendronate	401.11	.	pg/mL	BA	Alendronate 
1	2	1	6	Plasma	Alendronate	197.47	.	pg/mL	BA	Alendronate 
1	2	1	7	Plasma	Alendronate	122.63	.	pg/mL	BA	Alendronate 
1	2	1	8	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate 
1	2	1	10	Plasma	Alendronate	BLQ	100	pg/mL	BA	Alendronate 
```

sample output :
![image](https://github.com/user-attachments/assets/9752df95-df48-4345-a300-f0fc442a9a6c)
[Figure_NCA.docx](https://github.com/user-attachments/files/16333099/Figure_NCA.docx)

**Comment by billdenney** [comment]
The issue is that you don't have a time 0 measurement but the AUC is starting from time 0.

Can you please take a look at the data imputation vignette (https://billdenney.github.io/pknca/articles/v08-data-imputation.html) and let me know if that answers your question?  This is a common issue in PKNCA, so I want to get that documentation spot-on to help others.  Your feedback about difficulty with understanding it or lack of clarity would be helpful.

**Reply by jeslo328** [reply]
Hi Bill
Yes. I saw that later upon this post and tried the imputing 0. 
Another question though, shall the end be the 10 hours or inf. ?

intervals=data.frame(start=0, 
                                           end=10,

**Reply by billdenney** [reply]
I'm glad that the documentation helped.

The choice of end depends on the question you are looking to answer.  For most parameters with single dosing, you will want `Inf` as the end.  I use shorter intervals with single doses when I want to compare to results expected for the first dose of single dosing (usually aucint.last or aucint.inf.obs or similar).

**Reply by jeslo328** [reply]
Thank you so much Bill. The more i explore the more im learning and find PKNCA is very helpful and great.
Thanks for the prompt guidance too.
